packages/sanes-chrome-extension/src/context/PersonaProvider.tsx

```diff
@@ -47,24 +47,29 @@ export const PersonaProvider = ({ children, persona }: Props): JSX.Element => {
 
     chrome.runtime.onMessage.addListener((msg, sender, _sendResponse) => {
       const sameTarget = sender.id === chrome.runtime.id;
-      const msgToForeground = isMessageToForeground(msg);
-      const msgToPersonaProvider = msg.action === MessageToForegroundAction.TransactionsChanged;
-      if (!sameTarget || !msgToForeground || !msgToPersonaProvider) {
+      if (!sameTarget || !isMessageToForeground(msg)) {
         // Only handle messages from background script
         return;
       }
 
-      // TODO Refactor to do not hard check everything, if the message is not for here
-      // just ignore it. Change the above checks as well.
       switch (msg.action) {
         case MessageToForegroundAction.TransactionsChanged:
-          if (!Array.isArray(msg.data)) {
-            throw new Error('Data must be an array');
-          }
-          setTxs(msg.data);
+          getPersonaData()
+            .then(personaData => {
+              if (personaData) {
+                setTxs(personaData.txs);
+              } else {
+                // eslint-disable-next-line no-console
+                console.warn('Could not get persona data after receiving TransactionsChanged message');
+              }
+            })
+            .catch(error => {
+              // eslint-disable-next-line no-console
+              console.error(error);
+            });
           break;
         default:
-          throw new Error('Unknown action');
+        // this listener is not responsible for the given message, ignore message
```

Comment:
In this case do we need `default` at all?

Reply:
Good point. You are right, it is not strictly needed. But I found it useful to use it as a place for the comment. I think it is always good to be explicit about all possible cases. Before, we threw an Error but now we explicitly ignore other messages.